ops/terraform/modules/resources/bfd_pipeline/main.tf

```diff
@@ -29,6 +41,65 @@ resource "aws_security_group_rule" "allow_db_primary_access" {
   source_security_group_id  = aws_security_group.app.id   # The EC2 instance for the BFD Pipeline app.
 }
 
+# IAM policy and role to allow the BFD Pipeline read-write access to ETL bucket.
+#
+resource "aws_iam_policy" "bfd_pipeline_rif" {
+  name        = "bfd-${var.env_config.env}-pipeline-rw-s3-rif"
+  description = "Allow the BFD Pipeline application to read-write the S3 bucket with the RIF in it."
+
+  policy = <<EOF
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Sid": "BFD Pipeline RW S3 RIF: KMS",
+      "Action": ["kms:Decrypt"],
+      "Effect": "Allow",
+      "Resource": ["${data.aws_kms_key.master_key.arn}"]
+    },
+    {
+      "Sid": "BFD Pipeline RW S3 RIF: List Bucket",
+      "Action": ["s3:ListBucket"],
+      "Effect": "Allow",
+      "Resource": ["${data.aws_s3_bucket.rif.arn}"]
+    },
+    {
+      "Sid": "BFD Pipeline RW S3 RIF: Read Write Objects",
+      "Action": [
+        "s3:GetObject",
+        "s3:PutObject"
+      ],
+      "Effect": "Allow",
+      "Resource": ["${data.aws_s3_bucket.rif.arn}/*"]
+    }
+  ]
+}
+EOF
+}
+
+module "iam_profile_bfd_pipeline" {
+  source = "../iam"
+
+  env_config      = var.env_config
+  name            = "bfd_pipeline"
+}
+
+resource "aws_iam_role_policy_attachment" "bfd_pipeline_rif" {
+  role            = module.iam_profile_bfd_pipeline.role
+  policy_arn      = aws_iam_policy.bfd_pipeline_rif.arn
+}
+
+# Give the BFD Pipeline app read access to the Ansible Vault PW.
+#
+data "aws_iam_policy" "ansible_vault_pw_ro_s3" {
+  arn           = "arn:aws:iam::577373831711:policy/bfd-ansible-vault-pw-ro-s3"
+}
```

Comment:
Instead of hard coding the account-id in the `arn`, I think you should get it the passed in account_id variable.

Reply:
Done, though a `grep` indicates it's hardcoded in several other places, which we should address later.